### ISSUE TITLE
Is this giving false impression of authentication?

### ISSUE THREAD

**Issue by rajaraodv** [luisrudge/next.js-auth0#11]
If I am correct, entire JS is bundled and served in the client. So the 'secret' page is also served to the client browser. So if someone toggles the props, it will display the page! I think you are giving false impression of 'secure' page.

I think we  **must check it at the server before sending** the secure page.

**Comment by luisrudge** [comment]
The secret page is not included by default, since next.js loads new pages dynamically. But if the user manually adds a json object to the localStorage with the correct key, the app will mark the user as authenticated indeed. This is not solvable without having an external api to check if the token is valid. 
This project is just an example on how you'd secure your app with Auth0.

**Comment by rajaraodv** [comment]
I'm am saying if the user(or a hacker) manually goes to /secret page, then it Next will serve that page to the browser.

**Comment by luisrudge** [comment]
Yeah. You're right. Not sure how to fix that.

**Comment by pfortes** [comment]
@luisrudge I think this issue must be re-opened because everyone using this example as it is will be exposing all its secure content to anyone that is willing to change a prop using react dev tools.

I believe the correct approach is validating the users token always in _app and redirecting or not rendering any secure content if it is not valid.

There is a good example to look at:
https://github.com/bgold0/nextjs-auth-skeleton/blob/master/pages/_app.js

**Comment by luisrudge** [comment]
@pfortes you're right. this is pretty old and I was just playing around with auth0 and nextjs. Now that I'm working at Auth0 for almost 2 years, I can say that it is the wrong approach. Auth0 will be releasing guidance on this in the near future.

TLDR: treat nextjs apps as server apps.

I'll archive the repo for now.

**Comment by darkyen** [comment]
Anything that runs on the client side is exposed to a hacker or malicious agent being able to manipulate events in order to make the app render the client part, in an ideal app, you'll probably have a backend and the backend will have its own independent means of handling these application.s
